### ISSUE TITLE
IfcPolygonalFaceSet is not supported yet

### ISSUE THREAD

**Issue by CBenghi** [xBimTeam/XbimGeometry#106]
#### Expected behavior:
Meshing of most recent IFC4 classes should be generated

#### Actual behavior or exception details:
They are not.

####
Initial work on this has been carried out in Essentials
https://github.com/xBimTeam/XbimEssentials/issues/116

**Comment by martin1cerny** [comment]
Once in this @CBenghi, would you like to take a look at those from [IFC Alignment](http://www.buildingsmart-tech.org/ifc/IFC4x1/final/html/) as well?

```
IfcTriangulatedIrregularNetwork
IfcTransitionCurveSegment2D
IfcSectionedSolid
IfcSectionedSolidHorizontal
IfcOffsetCurveByDistances
IfcOrientationExpression
IfcOffsetCurveByDistances
IfcLinearPlacement
IfcCurveSegment2D
IfcCircularArcSegment2D
IfcAlignmentCurve
IfcAlignment2DVerticalSegment
IfcAlignment2DVertical
IfcAlignment2DVerSegParabolicArc
IfcAlignment2DVerSegLine
IfcAlignment2DVerSegCircularArc
IfcAlignment2DSegment
IfcAlignment2DHorizontalSegment
IfcAlignment2DHorizontal
```

:-)

**Comment by bekraft** [comment]
Is there any progress  here for the adaption of IFC4x1?

**Comment by martin1cerny** [comment]
Not yet.

**Comment by andyward** [comment]
Looking at this for v5 (soon). But may have to be a follow up release

**Comment by CBenghi** [comment]
Hi All,
I've implemented IFCPOLYGONALFACESET in the Maintenance/Xbim4.x branch.
Because it depends on a few changes on Essentials that need careful evaluation in xbim5 it might take a few weeks before it gets there, but it's on the way.

**Comment by andyward** [comment]
Looks like Steve has implemented in 8ebe65d198f3184ab82070dc8923d26368725d99

**Comment by SteveLockley** [comment]
IFCPOLYGONALFACESET is implemented in the geometry engine and in meshing as of version 5.0.219
